compaction/compaction.hh

```diff
@@ -85,12 +85,29 @@ struct compaction_data {
     }
 };
 
-struct compaction_result {
-    std::vector<sstables::shared_sstable> new_sstables;
+struct compaction_counters {
     std::chrono::time_point<db_clock> ended_at;
     uint64_t start_size = 0;
     uint64_t end_size = 0;
     uint64_t validation_errors = 0;
+
+    compaction_counters& operator+=(const compaction_counters& r) {
+        ended_at = r.ended_at;
+        start_size += r.start_size;
+        end_size += r.end_size;
+        validation_errors += r.validation_errors;
+        return *this;
+    }
+    friend compaction_counters operator+(const compaction_counters& l, const compaction_counters& r) {
+        auto tmp = l;
+        tmp += r;
+        return tmp;
+    }
+};
+
+struct compaction_result {
+    std::vector<sstables::shared_sstable> new_sstables;
+    compaction_counters counters;
```

Comment:
let's do this refactoring in a separate patch please so it'll stand out as a standalone change